python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -97,3 +103,301 @@ def _isBad(self, source):
+    secondary = pexConfig.Field(dtype=str, doc="Name of column with secondary flux measurement")
+    minimum = pexConfig.Field(dtype=float, optional=True, doc="Select objects with color greater than this")
+    maximum = pexConfig.Field(dtype=float, optional=True, doc="Select objects with color less than this")
+
+    def apply(self, catalog):
+        """Apply the color limit to a catalog
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Source to which to apply limits.
+
+        Returns
+        -------
+        selected : `numpy.ndarray`
+            Boolean array indicating whether each source is selected.
+        """
+        primary = lsst.afw.image.abMagFromFlux(catalog[self.primary])
+        secondary = lsst.afw.image.abMagFromFlux(catalog[self.secondary])
+        color = primary - secondary
+        selected = np.ones(len(catalog), dtype=bool)
+        if self.minimum is not None:
+            selected &= color > self.minimum
+        if self.maximum is not None:
+            selected &= color < self.maximum
+        return selected
+
+
+class FluxLimit(pexConfig.Config):
+    """Select sources using a flux limit
+
+    This object can be used as a `lsst.pex.config.Config` for configuring
+    the limit, and then the `apply` method can be used to test sources in
+    a catalog against the configured limit.
+    """
+    fluxField = pexConfig.Field(dtype=str, default="slot_CalibFlux_flux",
+                                doc="Name of the source flux field to use.")
+    minimum = pexConfig.Field(dtype=float, optional=True, doc="Select objects fainter than this flux")
+    maximum = pexConfig.Field(dtype=float, optional=True, doc="Select objects brighter than this flux")
+
+    def apply(self, catalog):
+        """Apply the magnitude limit to a catalog
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Source to which to apply limits.
+
+        Returns
+        -------
+        selected : `numpy.ndarray`
+            Boolean array indicating whether each source is selected.
+        """
+        flagField = self.fluxField + "_flag"
+        if flagField in catalog.schema:
+            selected = np.logical_not(catalog[flagField])
+        else:
+            selected = np.ones(len(catalog), dtype=bool)
+
+        flux = catalog[self.fluxField]
+        if self.minimum is not None:
+            selected &= flux > self.minimum
+        if self.maximum is not None:
+            selected &= flux < self.maximum
+        return selected
+
+
+class MagnitudeLimit(pexConfig.Config):
+    """Select sources using a magnitude limit
+
+    Note that this assumes that a zero-point has already been applied and
+    the fluxes are in AB fluxes. It is therefore principally intended for
+    reference catalogs rather than catalogs extracted from science images.
```

Comment:
Maybe make it explicit how flux (in Jansky!) is converted to magnitude in the docs, so the refcat vs. sourceCat distinction is clearer?

Reply:
Added mention of Jansky. I think questions about the specific implementation are best answered by the (simple) code.